### ISSUE TITLE
Missing functions in `consistent-function-scoping`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#931]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2446.61%20Funded-%2300A156.svg" />](https://oss.issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/931)

Arrow function (throws error):
```js
// ESLint: Move arrow function to the outer scope.(unicorn/consistent-function-scoping)
//
//                              here
//                              ⬇
export const someAction = () => (dispatch) =>
  dispatch({
    type: SOME_TYPE,
  });
```

Function declaration (no error):
```js
export function someAction() {
  return (dispatch) =>
    dispatch({
      type: SOME_TYPE,
    });
}
```

The arrow function in the second one should be reported.

----

Original posted by @Arttse https://github.com/sindresorhus/eslint-plugin-unicorn/issues/596#issuecomment-628711205

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $46.61)

- [<img src='https://avatars.githubusercontent.com/u/1402241?v=4' alt='fregante' width=24 height=24> fregante](https://oss.issuehunt.io/u/fregante) ($46.61)

### Submitted pull Requests
- [#2748 Fix inconsistent behavior in `consistent-function-scoping` rule](https://oss.issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/2748)
---

### Tips

- Checkout the [Issuehunt explorer](https://oss.issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://oss.issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by SashankBhamidi** [comment]
@fisker I’d like to work on this. Is the bounty still up for grabs? If yes, assign it to me and I’ll ship a PR.

**Comment by fregante** [comment]
It is

**Comment by SashankBhamidi** [comment]
@sindresorhus Could you please reopen this issue? It needs to be open for the bounty to be awarded to my PR on Issuehunt. Once the bounty is granted, the issue can be closed. As the maintainer, you can also award me the bounty since the post author hasn’t been responding. Thanks!
